src/guillotine/fetchContent.ts

```diff
@@ -991,10 +991,11 @@ export const fetchContent: ContentFetcher = buildContentFetcher<AdapterConstants
     componentRegistry: ComponentRegistry,
 });
 
-export function createResponse(props: FetchContentResult, context?: Context): GetStaticPropsResult<FetchContentResult> {
+export function createResponse(props: FetchContentResult, context?: Context, isStatic?: true): GetStaticPropsResult<FetchContentResult>;
+export function createResponse(props: FetchContentResult, context?: Context, isStatic?: false): GetServerSidePropsResult<FetchContentResult>;
+export function createResponse(props: FetchContentResult, context?: Context, isStatic = true): GetStaticPropsResult<FetchContentResult> | GetServerSidePropsResult<FetchContentResult> {
```

Comment:
This will only export a single function createResponse. 
Method overloading is not a thing in JS, only the last function will be exported.

Reply:
It is not a thing in JS, but it is one in TS, because it does compile-time checking. [Docs here](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads)

Reply:
Did not know 😮